From edfd598446cfd6d4348a3bfecd72b5665eef5bfc Mon Sep 17 00:00:00 2001
From: Akseli Lahtinen <akselmo@akselmo.dev>
Date: Mon, 16 Dec 2024 12:26:38 +0000
Subject: [PATCH] DolphinViewContainer: make sure searchbox is initialized
 before setSearchPath

If searchBox is not initialized, it can't be updated properly
with setSearchPath due to some buttons not existing.
There is also just no need to update it if it's not initialized.

Make sure we check for it's initialized during setting the search path.

BUG:497021


(cherry picked from commit 37b081331e19b8b10bc80b8aecc4ff43de2474c0)

fa289230 DolphinViewContainer: make sure searchbox is visible before setSearchPath
d0260e60 Check the initialization in setSearchPath

Co-authored-by: Akseli Lahtinen <akselmo@akselmo.dev>
---
 src/search/dolphinsearchbox.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/search/dolphinsearchbox.cpp b/src/search/dolphinsearchbox.cpp
index 56e6543e7..ee425501d 100644
--- a/src/search/dolphinsearchbox.cpp
+++ b/src/search/dolphinsearchbox.cpp
@@ -74,7 +74,7 @@ QString DolphinSearchBox::text() const
 
 void DolphinSearchBox::setSearchPath(const QUrl &url)
 {
-    if (url == m_searchPath) {
+    if (url == m_searchPath || !m_initialized) {
         return;
     }
 
-- 
2.47.1

